hydra-nextjs/README.md

```diff
@@ -0,0 +1,75 @@
+# ORY Hydra Next.js Reference Implementation
+
+This is a Next.js app which sets up a OAuth 2.0 and OpenID Connect Provider using [Ory Hydra](https://www.ory.sh/docs/hydra/) as a backend. It has an unstyled UI and doesn't implement user management but can be easily modified to fit into your existing system.
+
+# Features
+
+  * User login, logout, registration, consent
+  * CSRF protection with [edge-csrf](https://github.com/amorey/edge-csrf)
+  * Super-strict HTTP security headers (configurable)
+  * Client-side JavaScript disabled by default
+  * Unit tests with Jest
+  * E2E tests with Cypress
+  * Start/stop Hydra in development using docker-compose
+  * Easily customizable
+
+## Configuration
+
+This application can be configured using the following environment variables:
+
+| Name                    | Default                |
+| ----------------------- | ---------------------- |
+| SECURITY_HEADERS_ENABLE | false                  |
+| HYDRA_ADMIN_URL         | http://localhost:4445/ |
+
+## Development
+
+To install dependencies:
+
+```sh
+yarn install
+```
+
+To run the Next.js app server in development mode:
+
+```sh
+yarn dev
```

Comment:
When I run `yarn install` and `yarn dev` on my system (Debian), I see `localhost:3000` but it always displays "Login challenge is missing" and neither the sign-up nor sign-in page work. Am I missing some setup? If yes, can you please document it here?

Reply:
You need an oauth client to initiate the authentication flow and a running hydra server to provide a valid challenge. How would you recommend providing an oauth client for this project? Is there a docker image that I could use?